components/reusable-components/reusable-form/ReusableForm.tsx

```diff
@@ -0,0 +1,192 @@
+/* eslint-disable no-unused-vars */
+
+'use client';
+
+import React, { useState } from 'react';
+import * as Yup from 'yup';
+import { useFormik, FormikProps } from 'formik';
+import Link from 'next/link';
+import Image from 'next/image';
+import style from './reusableForm.module.scss';
+import TextInput from '../text-input/TextInput';
+import Button from '../button/Button';
+import github from '../../../public/icons-footer/github.svg';
+import linkedin from '../../../public/icons-footer/linkedIn.svg';
+import google from '../../../public/icons-footer/google.svg';
+import check from '../../../public/icons/check.svg';
+
+const ReusableForm = () => {
+  const [password, setPassword] = useState<string>('');
+
+  const initialValues = {
+    firstName: '',
+    lastName: '',
+    email: '',
+    password: '',
+  };
+
+  const validationSchema = Yup.object({
+    firstName: Yup.string()
+      .matches(/^[a-zA-Z .'-]+$/, '*Невалидно име!')
+      .min(2, '*Вашето име е премногу кратко!')
+      .max(50, '*Вашето име е премногу долго!')
+      .required('*Задолжително внесете име'),
+    lastName: Yup.string()
+      .min(2, '*Вашето презиме е премногу кратко!')
+      .max(50, '*Вашето презиме е премногу долго!')
+      .required('*Задолжително внесете презиме'),
+    email: Yup.string()
+      .email('*Невалидна емаил адреса')
+      .required('*Задолжително внесете електронка пошта'),
+    password: Yup.string()
+      .matches(/[a-zA-Z]/, '*Лозинката не ги исполнува сите услови!')
+      .min(8, '*Вашата лозинка е премногу кратка!')
+      .required('*Задолжително внесете лозинка'),
+  });
+
+  const formik: FormikProps<{
+    firstName: string;
+    lastName: string;
+    email: string;
+    password: string;
+  }> = useFormik({
+    initialValues,
+    validationSchema,
+    onSubmit: () => {},
+  });
+
+  const handlePasswordChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+    setPassword(value);
+    formik.handleChange(e);
+  };
```

Comment:
Here we will update the new state as well.
Here you will test the conditions and merge the new state with the previous state.
When testing the contitions you will check if the condition is met it will be true, if not it would be null. We need the null value here since the icon color should stay grey if the condition is not met. We are changing it to red on Submit.


About changing the icon color.
Since we have an svg icon, we can create a component that would have a color prop and will return the svg icon.
You will make the following change in the svg `stroke={color}`
and also the prop would have a default value(the gray color).

When we use that component here, we would have something like: 
`  <CheckPasswordValidityIcon color={getColor(valid)} />`

Reply:
You need a function that  will return a hex value, it will have an argument that can be boolean or null
if null return the grey color if it is true return the green if not the red